components/candle/challengeService.ts

```diff
@@ -1061,15 +1116,36 @@ export class ChallengeService extends ChallengeRegistry {
 
         const userData = getUserData(userId, gameVersion)
 
-        userData.Extensions.ChallengeProgression ??= {}
+        //ASSUMED: Challenges that are not global should always be completed
+        if (!challenge.Tags.includes("global")) {
+            userData.Extensions.ChallengeProgression ??= {}
 
-        userData.Extensions.ChallengeProgression[challenge.Id] ??= {
-            State: {},
-            Completed: false,
-            Ticked: false,
+            userData.Extensions.ChallengeProgression[challenge.Id] ??= {
+                State: {},
+                Completed: false,
+                Ticked: false,
+            }
+
+            userData.Extensions.ChallengeProgression[challenge.Id].Completed =
+                true
+        }
+
+        //Always count the number of completions
+        session.challengeContexts[challenge.Id].timesCompleted++
+
+        //If we have a Definition-scope with a Repeatable, we may want to restart it.
+        //TODO: Figure out what Base/Delta means. For now if Repeatable is set, we restart the challenge.
```

Comment:
At the start, the repeat counter starts at whatever the `Base` value is, every time the challenge is completed, the `Delta` value is then taken awake from the "repeat counter". We can probably use base to track this. I presume if the counter is less than 0, it won't activate. These are actually only really used on statistics, the rest of the time the `Base` is 1 with a `Delta` of 0.

Reply:
Good to know, thanks! I think it's okay to leave it for now then.